records/cache.go

```diff
@@ -0,0 +1,521 @@
+package records
+
+import (
+	"container/list"
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/emirpasic/gods/trees/redblacktree"
+	"github.com/segmentio/kafka-go"
+	"github.com/segmentio/kafka-go/protocol"
+	"github.com/segmentio/kafka-go/protocol/fetch"
+)
+
+// Key is a comparable type representing the identifiers that record batches
+// can be indexed by in a storage instance.
+type Key struct {
+	Addr            string
```

Comment:
Would a leader change have any effect here?

Reply:
It shouldn't, the intent is for this field to be the cluster address.